test/proxy.deltas.test.ts

```diff
@@ -0,0 +1,372 @@
+import {ProxyDeltasTestFactory} from "../contract-bindings/ethers";
+import {ProxyDeltasTest} from "../contract-bindings/ethers/ProxyDeltasTest";
+import buidler from "@nomiclabs/buidler";
+import {BigNumber, BigNumberish} from "ethers";
+import {assert} from "chai";
+import {submitFailing} from "./support";
+
+async function deployProxyDeltasTest(): Promise<ProxyDeltasTest> {
+  const [signer] = await buidler.ethers.getSigners();
+  return new ProxyDeltasTestFactory(signer)
+    .deploy()
+    .then((deltas) => deltas.deployed());
+}
+
+async function expectAddToDeltasWithInvalidCycleReverts(
```

Comment:
Personally, I’d inline this function to avoid unecessary indirection.

Reply:
It's used in 2 places. I've decided to make it DRY, because it contains some non-obvious code and hardcoded copies of strings from the contract. I don't feel very strong about keeping it like this.